torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2456,9 +2456,82 @@ Tensor svd_backward(const std::vector<torch::autograd::Variable> &grads, const T
     return u_term + sigma_term + v_term + imag_term;
   }
 
+
   return u_term + sigma_term + v_term;
 }
 
+Tensor _linalg_svd_rank_restricted_helper_backward(
+    const std::vector<torch::autograd::Variable>& grads,
+    const Tensor& input,
+    const Tensor& U,
+    const Tensor& S,
+    const Tensor& Vh,
+    const Tensor& rank,
+    const Tensor& unique_rank
+) {
+  auto grad = at::empty_like(input);
+
+  // If empty tensor in batch dimensions
+  if (!input.numel()) {
+    return grad;
```

Comment:
Better to return an undefined tensor.

Reply:
The `grad` tensor is not going to be undefined, but it is going to be empty. You can check that 'defined()' returns 'true' if the underlying 'impl_' pointer is not empty, and does not take into the consider the number of elements.